sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
That's true, but it will be exceedingly hard to miss that in code review, while the current thing is quite easy to miss. I'm really open to alternative suggestions here, but I do want to retain the property that using this type directly should be obviously wrong.

By the way, making this change made it much easier to port the code, because all the things I wanted to change stopped compiling.